### PR TITLE
Fix ValueNumber constant propagation with nested structs

### LINKED ISSUE #18672: Wrong constant propagation on Linux in release
On Linux in release the following program outputs `1` in debug, but `0` in release:
```csharp
using System.Runtime.CompilerServices;

struct S1
{
}

struct S3
{
    public int F1;
    public ulong F2;
    public S1 F6;
    public S3(int f1, S1 f6): this()
    {
        F1 = f1;
        F6 = f6;
    }
}

struct S5
{
    public S3 F0;
    public S5(S3 f0): this()
    {
        F0 = f0;
    }
}

public class Program
{
    public static void Main()
    {
        S5 vr13 = new S5(new S3(1, new S1()));
        int vr14 = vr13.F0.F1;
        M();
        System.Console.WriteLine(vr14); // 1 in debug, 0 in release
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    private static void M(){}
}
```

I'm not 100% sure that this isn't the same as #18259, however this only reproduces on Linux so there is definitely something different here.

### PR DESCRIPTION
**Pull request by briansull**
Fixes #18672

### CODE REVIEW

**Comment by briansull** [comment]
@AndyAyersMS  PTAL
/CC @dotnet/jit-contrib

**Comment by briansull** [comment]
There were no AsmDiffs with this change

**Comment by briansull** [comment]
This fix is ready to be merged
@AndyAyersMS PTAL
/CC @dotnet/jit-contrib

File: tests/src/JIT/Regression/JitBlue/GitHub_18672/GitHub_18672.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        S24W  s24W = new S24W(s24);
+        S24   s24X = s24;
+
+        int   fI2  = s24.FI2;
+        int   fI2W = s24W.FS24.FI2;
+        int   fI2X = s24X.FI2;
+
+        M();
+
+	if ((fI2 == 1) && (fI2W == 1) && (fI2X == 1))
```
Comment (AndyAyersMS) [inline tests/src/JIT/Regression/JitBlue/GitHub_18672/GitHub_18672.cs]: Nit: fix the indentation here

**Pull request merged**